### PR TITLE
Support search & minPriority in WebUI

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/brave-variations/issues/773

The PR adds to griffin.brave.com:
1. `minPriority` select field
2. `search` field

image

### CODE REVIEW

File: src/core/study_processor.ts [hunk 1]
```diff
@@ -39,20 +39,35 @@ export class StudyFilter {
...
 
+  get search(): string | undefined {
+    return this._search;
+  }
+
+  set search(value: string | undefined) {
+    this._search = value;
+    const sanitizedValue = value?.replaceAll(/\W/g, '');
+    this._searchRegexp =
+      value !== undefined ? new RegExp(`(${sanitizedValue})`, 'gi') : undefined;
```
[the remark this answers is not in the document]
Comment (atuchin-m) [reply src/core/study_processor.ts:57]: added sanitizing
@thypon @bcaller could you take a look?

File: src/web/app/app.tsx [hunk 2]
```diff
@@ -252,6 +284,58 @@ export function FilterCheckbox(props: {
   );
 }
 
+export function FilterPriority(props: {
```
Comment (petemill) [inline src/web/app/app.tsx:287]: nit: it is a filter ui control, so name it `PriorityFilter` (same with the search filter below)

File: src/web/css/style.css [hunk 3]
```diff
@@ -63,9 +68,28 @@ section.navbar {
...
   margin-right: 15px;
 }
 
+.filter-left {
+  margin-left: auto;
+  margin-right: 0px;
+}
+
+.filter-checkbox {
+  vertical-align:middle;
```
Comment (petemill) [inline src/web/css/style.css:84]: format: spacing

**Review by petemill — APPROVED** [review]

File: src/core/study_processor.ts [hunk 4]
```diff
@@ -39,20 +39,35 @@ export class StudyFilter {
...
   }
 
+  get search(): string | undefined {
+    return this._search;
+  }
+
+  set search(value: string | undefined) {
+    this._search = value;
+    const sanitizedValue = value?.replaceAll(/\W/g, '');
+    this._searchRegexp =
```
[the remark this answers is not in the document]
Comment (atuchin-m) [reply src/core/study_processor.ts:56]: Yes, getter & setter is only for optimization. 
`matches` is called multiple times, so it's a bad style to rebuild regex each time.

File: src/web/app/app.tsx [hunk 5]
```diff
@@ -56,21 +66,42 @@ export function FeatureList(props: {
   );
 }
 
-export function ExperimentItem(props: { exp: ExperimentModel }) {
+function maybeHightlight(filter: StudyFilter, text: string): JSX.Element {
```
[the remark this answers is not in the document]
Comment (atuchin-m) [reply src/web/app/app.tsx]: Done.

File: src/web/css/style.css (hunk 3 above)
Comment (atuchin-m) [reply src/web/css/style.css]: Fixed.

File: src/web/app/app.tsx (hunk 2 above)
Comment (atuchin-m) [reply src/web/app/app.tsx]: Done.

File: src/core/study_processor.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bcaller) [reply src/core/study_processor.ts:57]: Yes that looks good.

**Review by boocmp — APPROVED** [review]

**Pull request merged**